astropy/coordinates/representation.py

```diff
@@ -400,6 +529,85 @@ def transform(self, matrix):
         newxyz = u.Quantity(newxyz, oldxyz.unit, copy=False)
         return self.__class__(*newxyz, copy=False)
 
+    def _combine_operation(self, op, other, reverse=False):
+        try:
+            other_c = other.to_cartesian()
+        except Exception:
+            return NotImplemented
+
+        first, second = ((self, other_c) if not reverse else
+                         (other_c, self))
+        return self.__class__(*(op(getattr(first, component),
+                                   getattr(second, component))
+                                for component in first.components))
+
+    def mean(self, *args, **kwargs):
+        """Return the vector mean along a given axis.
+
+        Returns a new CartesianRepresentation instance with the means of the
+        x, y, and z components.
+
+        Refer to `numpy.mean` for full documentation of the arguments.
+        """
+        return self._apply('mean', *args, **kwargs)
+
+    def sum(self, *args, **kwargs):
+        """Return the vector sum along a given axis.
+
+        Returns a new CartesianRepresentation instance with the sums of the
+        x, y, and z components.
+
+        Refer to `numpy.sum` for full documentation of the arguments.
+        """
+        return self._apply('sum', *args, **kwargs)
+
+    def dot(self, other):
+        """Dot product of two representations.
+
+        Parameters
+        ----------
+        other : `~astropy.coordinates.BaseRepresentation`
+            If not already cartesian, it is converted.
+
+        Returns
+        -------
+        dot_product : `~astropy.units.Quantity`
+            The sum of the product of the x, y, and z components of ``self``
```

Comment:
Ah sorry, I was being blind and did not see the "of the product" words.  So never mind.